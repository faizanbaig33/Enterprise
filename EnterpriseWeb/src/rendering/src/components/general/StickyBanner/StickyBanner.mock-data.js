const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'StickyBanner',
  },
  fields: {
    componentSpacing: null,
    theme: {
      value: '',
    },
    sectionId: {
      value: '',
    },
    headlineText: {
      value: 'Sticky Banner Headline Text',
    },
    headlineLevel: {
      id: '7f10288a-beb1-4d95-a139-32df40094441',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3',
      name: 'Heading 3',
      displayName: 'Heading 3',
      fields: {
        Value: {
          value: 'h3',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    superscriptCTA: {
      value: {
        href: '',
      },
    },
    body: {
      value: 'Sticky Banner Body Copy Lorem Ipsum',
    },
    cta1Link: {
      value: {
        href: '/',
        text: 'View More',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
    },
    cta2Link: {
      value: {
        href: '/',
        text: 'View More',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta2Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'link',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
    },
  },
};

export default defaultData;
