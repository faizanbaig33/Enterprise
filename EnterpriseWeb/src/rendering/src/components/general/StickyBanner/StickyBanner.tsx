// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import classNames from 'classnames';
import { Component } from 'src/helpers/Component';
import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { StickyBannerTheme } from './StickyBanner.theme';
import { useState } from 'react';
import { SvgIcon } from 'src/helpers/SvgIcon';

export type StickyBannerProps = Feature.EnterpriseWeb.Components.General.StickyBanner.StickyBanner;
const StickyBanner = (props: StickyBannerProps) => {
  const { themeData } = useTheme(StickyBannerTheme);

  const [isDismissed, setIsDismissed] = useState(false);

  if (isDismissed) {
    return <></>;
  }

  return (
    <Component variant="full" dataComponent="general/stickybanner" {...props}>
      <div
        className={classNames(
          'theme-black fixed bottom-0 left-0 right-0 col-span-12 place-items-center bg-theme-bg'
        )}
      >
        <div className={classNames(themeData.classes.bannerWrapper)}>
          <div className={themeData.classes.textWrapper}>
            <Headline classes={themeData.classes.headline} {...props} />
            <BodyCopy classes={themeData.classes.bodyClass} {...props} />
          </div>
          <div className="col-span-2 place-content-center ml:col-span-5 ml:place-self-end ml:self-center">
            <ButtonGroup classes={themeData.classes.buttonGroupClass} {...props} />
          </div>
          <button
            className={classNames(themeData.classes.iconWrapper)}
            onClick={() => setIsDismissed(true)}
          >
            <SvgIcon icon="close" size="md" />
          </button>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<StickyBannerProps>(StickyBanner);
