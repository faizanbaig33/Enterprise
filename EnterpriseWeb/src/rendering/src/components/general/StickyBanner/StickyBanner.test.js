// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import StickyBanner from './StickyBanner';
import defaultData from './StickyBanner.mock-data';

it('renders correctly', () => {
  const component = snapshot(StickyBanner, { componentProps: defaultData });
  hasDataComponent(component, 'general/stickybanner');
});
