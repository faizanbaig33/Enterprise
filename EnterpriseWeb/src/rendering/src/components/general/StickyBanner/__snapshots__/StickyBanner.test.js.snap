// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/stickybanner"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s"
    >
      <div
        class="theme-black fixed bottom-0 left-0 right-0 col-span-12 place-items-center bg-theme-bg"
      >
        <div
          class="grid-rows-auto grid grid-cols-2 place-items-center px-m ml:px-l lg:px-m ml:mx-auto ml:max-w-screen-lg min-h-[75px]  ml:grid-cols-12 ml:place-items-start my-s"
        >
          <div
            class="col-span-2 text-center ml:col-span-7 ml:text-left mb-m ml:mb-0 self-center"
          >
            <div
              class="items-top font-sans  text-white text-sm-m ml:text-m  font-heavy"
            >
              <h3>
                Sticky Banner Headline Text
              </h3>
              
            </div>
            <div
              class="text-white text-body font-regular mt-xxs"
            >
              <div
                class="body-copy"
              >
                <div>
                  Sticky Banner Body Copy Lorem Ipsum
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-span-2 place-content-center ml:col-span-5 ml:place-self-end ml:self-center"
          >
            <div
              class="flex-col items-center font-heavy mb-0 gap-y-6 ml:gap-0 mb-s flex items-start md:flex-row md:items-center md:space-x-4"
            >
              <a
                class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray mr-0  mr-m"
                href="/"
                role="link"
                tabindex="0"
                target=""
                title=""
              >
                View More
                <span
                  class="ml-xxs"
                />
              </a>
              <a
                class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray ml:ml-0 text-white"
                href="/"
                role="link"
                tabindex="0"
                target=""
                title=""
              >
                View More
                <span
                  class="ml-xxs"
                />
              </a>
            </div>
          </div>
          <button
            class="absolute top-3 right-3 text-white"
          >
            <span />
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
